src/modules/client/hooks/useClientAlerts.tsx

```diff
@@ -0,0 +1,80 @@
+import { useMemo } from 'react';
+import { clientBriefName } from '@/modules/hmis/hmisUtil';
+import { useHasRootPermissions } from '@/modules/permissions/useHasPermissionsHooks';
+import {
+  ClientWithAlertFieldsFragment,
+  useGetHouseholdClientAlertsQuery,
+} from '@/types/gqlTypes';
+
+enum AlertPriority {
+  high = 3,
+  medium = 2,
+  low = 1,
+}
+
+interface ClientAlertParams {
+  householdId?: string;
+  client?: ClientWithAlertFieldsFragment;
+}
+
+export default function useClientAlerts(
+  params: ClientAlertParams,
+  showClientName?: boolean
+) {
+  const [canViewClientAlerts] = useHasRootPermissions(['canViewClientAlerts']);
+
+  const {
+    data: { household } = {},
+    loading,
+    error,
+  } = useGetHouseholdClientAlertsQuery({
+    variables: { id: params.householdId || '' },
+    skip: !params.householdId || !canViewClientAlerts,
+  });
+
+  const canViewAlertsForAnyHouseholdMembers = useMemo(() => {
+    if (!canViewClientAlerts) return false;
+    if (!household) return true;
+    const clients = household.householdClients.map((c) => c.client);
+    return clients.some((c) => c.access.canViewClientAlerts);
+  }, [household, canViewClientAlerts]);
+
+  const clientAlerts = useMemo(() => {
+    const clients = household
+      ? household.householdClients.map((c) => c.client)
+      : params.client
+      ? [params.client]
+      : [];
+
+    const clientAlerts = clients.flatMap((c) =>
+      c.alerts.map((a) => {
+        return {
+          alert: a,
+          clientName: clientBriefName(c),
+          showClientName: showClientName,
+        };
+      })
+    );
```

Comment:
I think this looks good! Would prefer multi-char variable names though:

```suggestion
    const clientAlerts = clients.flatMap((client) =>
      client.alerts.map((alert) => ({
        alert,
        clientName: clientBriefName(client),
        showClientName,
      }))
    );
```